apsuite/commisslib/measure_tbt_data.py

```diff
@@ -0,0 +1,310 @@
+        return state
+
+    def prepare_timing(self, state=None):
+        """."""
+        super().prepare_timing(state)  # BPM trigger timing
+        # magnets trigger timing below
+        prms = self.params
+        trigpingh = self.devices['trigpingh']
+        trigpingh.source = state.get('trigpingh_source', prms.timing_event)
+        trigpingh.nr_pulses = state.get('trigpingh_nrpulses',
+                                        prms.trigpingh_nrpulses)
+        dly = state.get('trigpingh_delay', prms.trigpingh_delay)
+        if dly is not None:
+            trigpingh.delay = dly
+
+        trigpingv = self.devices['trigpingv']
+        trigpingv.source = state.get('trigpingv_source', prms.timing_event)
+        trigpingh.nr_pulses = state.get('trigpingv_nrpulses',
+                                        prms.trigpingv_nrpulses)
+        dly = state.get('trigpingv_delay', prms.trigpingv_delay)
+        if dly is not None:
+            trigpingv.delay = dly
+
+    def get_magnets_strength(self):
+        """."""
+        return self.devices['pingh'].strength, self.devices['pingv'].strength
+
+    def set_magnets_strength(self, hkick=None, vkick=None):
+        """."""
+        pingh, pingv = self.devices['pingh'], self.devices['pingv']
+        if hkick is not None:
+            pingh.strength = hkick
+        else:
+            pingh.strength = self.params.hkick / 1e6
+        _time.sleep(0.5)
+        if vkick is not None:
+            pingv.strength = vkick
+        else:
+            pingv.strength = self.params.vkick / 1e6
+        _time.sleep(0.5)
+
+    def do_measurement(self):
+        """."""
+        currinfo = self.devices['currinfo']
+        init_timing_state = self.get_timing_state()
+        init_magnets_strength = self.get_magnets_strength()
+        current_before = currinfo.current()
+        self.prepare_timing()
+        self.set_magnets_strength()  # gets strengths from params
+        self.data['measurement_error'] = False  # error flag
+        try:
+            self.acquire_data()  # BPMs signals + relevant info are acquired
+                                 # such as timestamps tunes, stored current
+                                 # rf frequency, acq rate, nr samples, etc.
+        except Exception as e:
+            print(f'An error occurred during acquisition: {e}')
+            self.data['measurement_error'] = True
+        self.recover_timing_state(init_timing_state)
+        self.set_magnets_strength(init_magnets_strength)  # restore strengths
+        self.data['current_before'] = current_before
+        self.data['current_after'] = self.data.pop('stored_current')
+
+    def get_default_fname(self):
+        """."""
+        hkick, vkick = self.params.hkick, self.params.vkick
+        tm = self.data['timestamp']
+        fmt = '%Y-%m-%d-%H-%M-%S'
+        tmstp = _datetime.datetime.fromtimestamp(tm).strftime(fmt)
+        stg = f'hkick_{int(round(hkick)):3d}_vkick_{int(round(vkick)):3d}'
```

Comment:
Add information regarding which kicks were kicked.

Reply:
This info will be conveyed by the `"hkick"` or `"vkick"` strings and their corresponding values in the default fname. If only one of them is present, then only the corresponding magnet has pulsed. If both are present, both magnets pulsed.

Reply:
I liked the idea of embedding this info in the `hkick` and `vkick` arguments, but maybe it would be better to always keep both on the name. The kicker that didn't pulse could have a different value, though. I was thinking on something like `none` or `nopulse` or `standby` or `inactive` or something else. From these suggestions, I think I prefer `inactive`, but I think you will come up with something better :smile: